pkg/os/shell/shell_windows.go

```diff
@@ -67,32 +91,18 @@ func detect() (string, error) {
 		if err != nil {
 			return "cmd", err // defaulting to cmd
 		}
-		switch {
-		case strings.Contains(strings.ToLower(shell), "powershell"):
-			return "powershell", nil
-		case strings.Contains(strings.ToLower(shell), "pwsh"):
-			return "powershell", nil
-		case strings.Contains(strings.ToLower(shell), "cmd"):
-			return "cmd", nil
-		default:
-			shell, _, err := getNameAndItsPpid(shellppid)
-			if err != nil {
-				return "cmd", err // defaulting to cmd
-			}
-			switch {
-			case strings.Contains(strings.ToLower(shell), "powershell"):
-				return "powershell", nil
-			case strings.Contains(strings.ToLower(shell), "cmd"):
-				return "cmd", nil
-			default:
-				return "cmd", nil // this could be either powershell or cmd, defaulting to cmd
-			}
-		}
+		return analyzeShell(shell, shellppid)
 	}
 
 	if os.Getenv("__fish_bin_dir") != "" {
 		return "fish", nil
 	}
 
-	return filepath.Base(shell), nil
+	baseShell := filepath.Base(shell)
+
+	if baseShell != "powershell" && baseShell != "cmd" {
+		return analyzeShell(shell, uint32(os.Getpid()))
```

Comment:
You will need something similar to the code at the beginning of `if shell == ""` to avoid the lint failures:
```
		pid := os.Getppid()
		if pid < 0 || pid > math.MaxUint32 {
			return "", fmt.Errorf("integer overflow for pid: %v", pid)
		}
```

If we need `pid` in multiple places, we can get it at the beginning of  this function.

Reply:
Yes, thanks for review